src/components/application_manager/test/commands/hmi/navi_audio_start_stream_request_test.cc

```diff
@@ -0,0 +1,378 @@
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_))
+      .WillOnce(Return(MockAppPtr()));
+  EXPECT_CALL(mock_app_manager_, HMILevelAllowsStreaming(_, _)).Times(0);
+  EXPECT_CALL(*mock_app_, set_audio_streaming_approved(_)).Times(0);
+
+  Event event(hmi_apis::FunctionID::Navigation_StartAudioStream);
+  event.set_smart_object(*msg_);
+
+  command_->on_event(event);
+}
+
+TEST_F(AudioStartStreamRequestTest,
+       OnEvent_Navigation_StopAudioStream_SUCCESS) {
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  Event event(hmi_apis::FunctionID::Navigation_StopAudioStream);
+
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(mock_app_manager_, HMILevelAllowsStreaming(_, _)).Times(0);
+
+  command_->on_event(event);
+}
+
+TEST_F(AudioStartStreamRequestTest, OnEvent_InvalidApp_Cancelled) {
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  InitBasicMessage();
+  (*msg_)[strings::params][hmi_response::code] =
+      hmi_apis::Common_Result::REJECTED;
+
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(MockAppPtr()));
+  EXPECT_CALL(mock_app_manager_, HMILevelAllowsStreaming(_, _)).Times(0);
+  EXPECT_CALL(*mock_app_, set_audio_streaming_approved(_)).Times(0);
+
+  Event event(hmi_apis::FunctionID::Navigation_StartAudioStream);
+  event.set_smart_object(*msg_);
+
+  command_->on_event(event);
+}
+
+TEST_F(AudioStartStreamRequestTest, OnTimeOut_Terminated) {
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  InitBasicMessage();
+  (*msg_)[strings::params][strings::connection_key] = 3u;
+
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(MockAppPtr()));
+
+  EXPECT_CALL(mock_app_manager_, TerminateRequest(3u, kCorrelationID));
+
+  command_->onTimeOut();
+}
+
+TEST_F(AudioStartStreamRequestTest,
+       RetryStartSession_AudioStreamApproved_SUCCESS) {
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).WillOnce(Return(true));
+  const uint32_t retry_number_null = 0u;
+  EXPECT_CALL(*mock_app_, set_audio_stream_retry_number(retry_number_null));
```

Comment:
@indlu, why do you expect that `retry_number_null` will be `0`? This condition looks unclear.